src/rule_definition_tools.jl

```diff
@@ -59,6 +137,9 @@ For examples, see ChainRulesCore' `rules` directory.
 See also: [`frule`](@ref), [`rrule`](@ref), [`AbstractRule`](@ref)
 """
 macro scalar_rule(call, maybe_setup, partials...)
+    ############################################################################
+    # Setup: normalizing input form etc
+
```

Comment:
Can this be broken up into functions? I'd love for this to not be 100 lines long...

Reply:
i still feel the same

But if it is not easy / does not make sense to you, @oxinabox , then that's fine by me too

Reply:
Wahooo! Thanks 🎉